src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -144,13 +146,17 @@ impl CatalogState {
         }
     }
 
-    pub(super) fn pack_role_update(&self, id: RoleId, diff: Diff) -> Option<BuiltinTableUpdate> {
+    pub(super) fn pack_role_update(
+        &self,
+        id: RoleId,
+        diff: Diff,
+    ) -> Option<Vec<BuiltinTableUpdate>> {
```

Comment:
Returning just a `Vec<BuiltinTableUpdate>` may simplify this API a bit. The `PUBLIC` role can just return `Vec::new()` instead of `None`. Pretty subjective though, so up to you.

Reply:
This is a good call, I update it!